pkg/scheduler/api/job_info.go

```diff
@@ -228,6 +233,22 @@ func (ti *TaskInfo) ClearLastTxContext() {
 	ti.LastTransaction = nil
 }
 
+// Return if the pod of a task is scheduling gated by checking if length of sch gates is zero
+// When the Pod is not yet created or sch gates field not set, return false
+func calSchedulingGated(pod *v1.Pod) bool {
+	// Only enable if features.PodSchedulingReadiness feature gate is enabled
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodSchedulingReadiness) {
+		return pod != nil && pod.Spec.SchedulingGates != nil && len(pod.Spec.SchedulingGates) != 0
+	}
+	return false
+
+}
+
+// ti.SchGated is initialized in NewTaskInfo
+func (ti *TaskInfo) SchedulingGated() bool {
+	return ti.SchGated
```

Comment:
Should this filed be exported?

Reply:
With this function, there is no need to export it. I will change it to `schGated`.

Reply:
I would like to use this filed instead of a function calling.

Reply:
It's an optional way, but seems there is no need to expose an extra filed of task.

Reply:
I think it's reasonable.